logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt

```diff
@@ -277,4 +316,52 @@ class MLSConversationDataSource(
         conversationDAO.getAllMembers(conversationID).first().map { idMapper.fromDaoModel(it.user) }
     }
 
+    private suspend fun retryOnCommitFailure(groupID: GroupID, operation: suspend () -> Either<CoreFailure, Unit>) =
+        operation()
+            .flatMapLeft {
+                handleCommitFailure(it, groupID, operation)
+            }
+
+    private suspend fun handleCommitFailure(
+        failure: CoreFailure,
+        groupID: GroupID,
+        retryOperation: suspend () -> Either<CoreFailure, Unit>
+    ): Either<CoreFailure, Unit> {
+        return when (failure.getStrategy()) {
+            CommitStrategy.KEEP_AND_RETRY -> keepCommitAndRetry(groupID)
```

Comment:
do we have any documentation about when we should retry, abort or discard and retry? I'm curious about future operation that how can we figure out what strategy we should pick for them!

Reply:
I'm implementing this https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/614236365/Use+case+send+a+commit+message+MLS